code/modules/jobs/job_types/shaft_miner.dm

```diff
@@ -71,7 +71,7 @@
 	for(var/obj/item/gps/G in H.contents)
 		G.gpstag = "MINE[gps_number]"
 		gps_number ++
-	if (H.client?.prefs.miner_dwarf)
+	if (H.client?.prefs.miner_dwarf && prob(15))
 		H.dna?.add_mutation(DWARFISM, MUT_EXTRA)
```

Comment:
not entirely sure because i've not read about it
but from the way i've seen it used, i assume it's a way to assert the existence of a specific variable before it attempts to do something with it

Reply:
> what does the question mark do

Null check. Basically short form if(H.client)

Reply:
fukkin noobs